DesignDoc/RMA.md

```diff
@@ -1,3 +1,205 @@
+1. Consumer request techincial dispatch/schedule if needed
+1. Consumer ships the defective part for failure analysis, and notify the consumer about the shipment status of the defective parts
+1. Publisher notifies the failure analysis report to consumer
+## Out of scope
+
+We would not be discussing the transport methodologies preferred for this notification message.
+
+## Stages of RMA Lifecycle
+
+### Creation
+
+At this stage RMA(s) is raised by the Consumer, this might include:
+
+1. Collect the information, including 
+   * device/part basic information, e.g., serial number, part type, OS software version
+   * device/part alarms and logs
+   * site information where the defective parts resides
+   * contact information from Consumer side
+   * RMA service level, e.g., normal / escalate service
+
+1. Consumer request for RMA from Publisher
+1. Request type for each customer (email, API, manual).
+
+### Review/Approval 
+
+At this stage, Publisher will validate the required information for the RMA case, and approve / reject / request more information, which may include:
+   * whether any mandotary information is missing
+   * whether a kind of service level is supported for specific region
+
+### Parts tracking 
+
+There are two seperate flows at this stage:
+   1. Replacement shippment:
+      1. Publisher will provide the shippment information, including carrier, planned arrive date 
+      1. Publisher will notify the status of the replacement shippment and delivery
+   2. Defective part shippment:
+      1. Consumer will provide the shippment information, including carrier, planned arrive date 
+      1. Publisher will notify the status of the replacement shippment and delivery
+
+### Notification
+
+One RMA case is dedicated to one part with one unique identifier, e.g., serial number. Each RMA involves multiple communication messages between the publisher and consumer. Each RMA case has a unique ID.
+
+## Notification structure
+
+### General format
+
+`field_name:field_type:description`
+
+### Field Details
+
+#### case_id
+
+Identifier for a RMA case which dedicated to a part with a unique identifier. 
+
+`case_id:string:"RMA case identifier"`
+
+
+#### case_status
+
+RMA case status 
+`case_status:enum value:"stage of RMA case" enumeration: [OPEN, APPROVED, REJECTED]`
+
+
+#### case_status_time
+
+Epoch time when the status is updated from Publisher, we have one specific time for each status listed in `case_status`
+
+`case_open_time:date time:"Unix epoch time when consumer starts to request the RMA"`
+`case_approved_time:date time:"Unix epoch time when publisher approve to request the RMA"`
+`case_rejected_time:date time:"Unix epoch time when publisher reject the RMA"`
+
+#### part_shippment_status 
+
+Replacement / defective part shippment status
+
+`service_type:enum value:"Type of service: leased wave/dark fiber" enumeration: [LEASED_WAVE, DARK_FIBER]`
+
+#### part_type
+
+Replacement or defective parts
+
+#### service_impact
```

Comment:
Fields from line 99-116 are for fiber maintenance notification, I don't think they add value to a RMA related notification.

Reply:
Removed as suggested. For RMA, we may still need the service impacted which is included in the problem description section now.